docs/debugging-allocations.md

```diff
@@ -0,0 +1,187 @@
+```
+
+which this kind of block repeated for each allocation counter test. But let's go through each line. The first line is the name of the allocation test, most relevant is the file name (`test_future_lots_of_callbacks.swift`). So we seem to be testing how many allocations futures with many callbacks are doing.
+
+```
+- /Users/johannes/devel/swift-nio/IntegrationTests/tests_04_performance/test_01_resources/test_future_lots_of_callbacks.swift
+```
+
+Next, we see
+
+```
+test_future_lots_of_callbacks.remaining_allocations: 0
+test_future_lots_of_callbacks.total_allocations: 75001
+test_future_lots_of_callbacks.total_allocated_bytes: 4138056
+```
+
+which are the aggregates values: At the end of the test, there are `remaining_allocations: 0` which means that this allocation test didn't leak, good! Then, we see
+
+    test_future_lots_of_callbacks.total_allocations: 75001
+    
+Which means that at the end of the test, we saw 75001 allocations in total. As a rule, we usually run the workload of every allocation test 1000 times. That means, you want to divide 75001 by the 1000 runs. In other words: Each iteration of this allocation test allocated 75 times. The extra one allocation is just some noise that we have to ignore. In many (especially multi-threaded test cases) there is some noise which is why we run the workload usually 1000 times which makes it easy to tell the signal from the noise.
+
+Finally, we see
+
+    test_future_lots_of_callbacks.total_allocated_bytes: 4138056
+
+Which is the total number of bytes allocated by this test.
+
+
+Below, we see
+
+```
+DEBUG: [["remaining_allocations": 0, "total_allocations": 75001, "total_allocated_bytes": 4138056], ["total_allocations": 75001, "remaining_allocations": 0, "total_allocated_bytes": 4138056], ["total_allocated_bytes": 4138056, "total_allocations": 75001, "remaining_allocations": 0], ["total_allocated_bytes": 4138056, "total_allocations": 75001, "remaining_allocations": 0], ["remaining_allocations": 0, "total_allocations": 75001, "total_allocated_bytes": 4138056], ["total_allocations": 75001, "remaining_allocations": 0, "total_allocated_bytes": 4138056], ["total_allocated_bytes": 4138056, "total_allocations": 75001, "remaining_allocations": 0], ["total_allocations": 75001, "total_allocated_bytes": 4138056, "remaining_allocations": 0], ["remaining_allocations": 0, "total_allocations": 75001, "total_allocated_bytes": 4138056], ["total_allocations": 75001, "total_allocated_bytes": 4138056, "remaining_allocations": 0]]
```

Comment:
Is the `DEBUG:` just so we can identify that line so we can validate the numbers? Might be worth mentioning it's not related to the build configuration.

Reply:
good point, added